Modifying Questionnaire to Emotion Survey

carolinentran opened the pull request:
modified social support questionnaire with emotion survey

# Task 4: Add a questionnaire 

## :recycle: Current situation & Problem
The application currently has a Social Support Questionnaire. To make it more relevant to the Behavior Project, I added different questions regarding emotions, feeling of accomplishments, and actual goals completed. The questionnaire also includes conditional questions (depending on the user's answers on their emotions and challenges).


## :gear: Release Notes 
- User is reminded to take daily survey 
- Questionnaire opens as a screen/modal 
- Questionnaire allows user to close it and come back at a later time 
- Questionnaire entry button is disabled if user already filled out the daily form 


## :books: Documentation
- Reference for HL7 FHIR: https://hl7.org/fhir/R4/questionnaire.html
- Created new questionnaire using Phoenix Survey Builder: https://stanfordbdhg.github.io/phoenix/ 
- Debugging discussion: https://github.com/orgs/CS342/discussions/47


## :white_check_mark: Testing
- While testing my modifications, I found that the xcode simulator was not loading my changes. The modified questionnaire does show up when running the iPhone simulator though so this would be the best way to test the changes made. 

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [ ] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

## Screenshots of Questionnaire: 

![simulator_screenshot_1FE943DA-60EB-46B4-85BE-2AAE053CA283](https://github.com/CS342/2024-Behavior/assets/89322465/09731e5b-6ffe-4095-a797-3a237aff8205)
![simulator_screenshot_F6F642D1-8D69-4D6F-AF38-4DBE25DE29CB](https://github.com/CS342/2024-Behavior/assets/89322465/5233b6aa-2eba-4dfa-b3e8-6aaff03fa058)

carolinentran committed:
Modifying Questionnaire to Emotion Survey

modified social support questionnaire with emotion survey

carolinentran committed:
Adding comments

added comments for replacement of questionnaire content

carolinentran committed:
Build Fix: File Path

There was a path error with how I added the new json file. As a quick fix, I replaced the old questionnaire json with the new json. I'd like to discuss this more with the teaching staff for future improvements.

carolinentran committed:
Delete EmotionsQuestionnaireTask4.json

removing this since I modified the social support json directly

carolinentran committed:
Merge branch 'main' into feature/task4-addAQuestionnaire

vishnuravi reviewed (changes requested):
Hi @carolinentran,

Great job on this task! There are a couple of small things that could be improved:

1. The questionnaire file is still called `SocialSupportQuestionnaire.json`. I would change this to match your survey, e.g. `EmotionSurvey.json` and then update the references in your code.
2. The title on each question view is still `Social Support`. I would update this to be `Emotion Survey`.

Nice job incorporating some navigation logic into the survey JSON! It seems like there is an issue with the `enableWhen` statements for a couple of the questions. Although this is not required for the assignment, I'm happy to go over this and help you fix the bug.

Great work overall!

carolinentran committed:
Addressed File-Renaming Feedback

I changed the title of the questionnaire and other labeling in the json file. I also renamed the json file itself to better match the questionnaire.

carolinentran committed:
Added Copyright and Licensing File

carolinentran committed:
solved merge conflicts

vishnuravi reviewed (approved):
Nicely done @carolinentran!
